packages/parser/src/schemas/cognito.ts

```diff
@@ -0,0 +1,522 @@
+ *     "userAttributes": { "string": "string" },
+ *     "scopes": [ "string", "string" ],
+ *     "groupConfiguration": {
+ *       "groupsToOverride": [ "string", "string" ],
+ *       "iamRolesToOverride": [ "string", "string" ],
+ *       "preferredRole": "string"
+ *     },
+ *     "clientMetadata": { "string": "string" }
+ *   },
+ *   "response": {
+ *     "claimsAndScopeOverrideDetails": {
+ *       "idTokenGeneration": {
+ *         "claimsToAddOrOverride": { "string": "string" },
+ *         "claimsToSuppress": [ "string", "string" ]
+ *       },
+ *       "accessTokenGeneration": {
+ *         "claimsToAddOrOverride": { "string": "string" },
+ *         "claimsToSuppress": [ "string", "string" ],
+ *         "scopesToAdd": [ "string", "string" ],
+ *         "scopesToSuppress": [ "string", "string" ]
+ *       },
+ *       "groupOverrideDetails": {
+ *         "groupsToOverride": [ "string", "string" ],
+ *         "iamRolesToOverride": [ "string", "string" ],
+ *         "preferredRole": "string"
+ *       }
+ *     }
+ *   }
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html}
+ */
+export const PreTokenGenerationTriggerSchemaV2 = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    scopes: z.array(z.string()),
+    groupConfiguration: z.object({
+      groupsToOverride: z.array(z.string()),
+      iamRolesToOverride: z.array(z.string()),
+      preferredRole: z.string(),
+    }),
+    clientMetadata: z.record(z.string(), z.string()),
+  }),
+  response: z.object({
+    claimsAndScopeOverrideDetails: z.object({
+      idTokenGeneration: z.object({
+        claimsToAddOrOverride: z.record(z.any()),
+        claimsToSuppress: z.array(z.string()),
+      }),
+      accessTokenGeneration: z.object({
+        claimsToAddOrOverride: z.record(z.any()),
+        claimsToSuppress: z.array(z.string()),
+        scopesToAdd: z.array(z.string()),
+        scopesToSuppress: z.array(z.string()),
+      }),
+      groupOverrideDetails: z.object({
+        groupsToOverride: z.array(z.string()),
+        iamRolesToOverride: z.array(z.string()),
+        preferredRole: z.string(),
+      }),
+    }),
+  }),
+});
+
+/**
+ * A zod schema for a Cognito Migrate User trigger event.
+ *
+ * @example
+ * ```json
+ * {
+ *   "userName": "string",
+ *   "request": {
+ *     "password": "string",
+ *     "validationData": { "key": "value" },
+ *     "clientMetadata": { "key": "value" }
+ *   },
+ *   "response": {
+ *     "userAttributes": { "key": "value" },
+ *     "finalUserStatus": "string",
+ *     "messageAction": "string",
+ *     "desiredDeliveryMediums": [ "string" ],
+ *     "forceAliasCreation": true,
+ *     "enableSMSMFA": true
+ *   }
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-migrate-user.html}
+ */
+export const MigrateUserTriggerSchema = z.object({
```

Comment:
This is missing several fields, and the response should have values as `null`, see example:

```json
{
  "version": "1",
  "triggerSource": "UserMigration_Authentication",
  "region": "eu-west-1",
  "userPoolId": "eu-west-1_aaBBcC8dd",  
  "userName": "example@email.com",
  "callerContext": {
    "awsSdkVersion": "aws-sdk-unknown-unknown",
    "clientId": "1ab23kqn9j6ncsimddqs13sgok"
  },
  "request": {
    "password": "Password1!",
    "validationData": null,
    "userAttributes": null
  },
  "response": {
    "userAttributes": null,
    "forceAliasCreation": null,
    "enableSMSMFA": null,
    "finalUserStatus": null,
    "messageAction": null,
    "desiredDeliveryMediums": null
  }
}
```